src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,463 @@
+        return tooltipRenderer
+          ? tooltipRenderer(val, step)
+          : formatNumberValue(val, step, fractionDigits);
+      },
+      [fractionDigits, step, tooltipRenderer],
+    );
+
+    const getAllowableThumbRange = React.useCallback(
+      (index: number) => {
+        if (thumbMode === 'inhibit-crossing') {
+          const minVal = index === 0 ? min : currentValues[index - 1] + step;
+          const maxVal =
+            index < currentValues.length - 1
+              ? currentValues[index + 1] - step
+              : max;
+          return [minVal, maxVal];
+        }
+        return [min, max];
+      },
+      [max, min, step, thumbMode, currentValues],
+    );
+
+    const [activeThumbIndex, setActiveThumbIndex] = React.useState<
+      number | undefined
+    >(undefined);
+
+    const updateThumbValue = React.useCallback(
+      (
+        event: PointerEvent,
+        callback?: (values: ReadonlyArray<number>) => void,
+      ) => {
+        if (containerRef.current && undefined !== activeThumbIndex) {
+          const percent = getPercentageOfRectangle(
+            containerRef.current.getBoundingClientRect(),
+            event.pageX,
+          );
+          let pointerValue = min + (max - min) * percent;
+          pointerValue = roundValueToClosestStep(pointerValue, step, min);
+          const [minVal, maxVal] = getAllowableThumbRange(activeThumbIndex);
+          pointerValue = getBoundedValue(pointerValue, minVal, maxVal);
+          if (pointerValue === currentValues[activeThumbIndex]) {
+            return;
+          }
+          const newValues = [...currentValues];
+          newValues[activeThumbIndex] = pointerValue;
+          setCurrentValues(newValues);
+          callback?.(newValues);
+        }
+      },
+      [activeThumbIndex, min, max, step, getAllowableThumbRange, currentValues],
+    );
+
+    const handlePointerMove = React.useCallback(
+      (event: PointerEvent): void => {
+        updateThumbValue(event, onUpdate);
+      },
+      [onUpdate, updateThumbValue],
+    );
+
+    // function called by Thumb keyboard processing
+    const onThumbValueChanged = React.useCallback(
+      (index: number, value: number) => {
+        if (currentValues[index] === value) {
+          return;
+        }
+        const newValues = [...currentValues];
+        newValues[index] = value;
+        setCurrentValues(newValues);
+        onChange?.(newValues);
+      },
+      [currentValues, onChange],
+    );
+
+    const onThumbActivated = React.useCallback((index: number) => {
+      setActiveThumbIndex(index);
+    }, []);
+
+    const handlePointerUp = React.useCallback(
+      (event: PointerEvent) => {
+        updateThumbValue(event, onChange);
+        setActiveThumbIndex(undefined);
+        event.preventDefault();
+        event.stopPropagation();
+      },
+      [onChange, updateThumbValue],
+    );
+
+    const handlePointerDownOnSlider = React.useCallback(
```

Comment:
I think you need to focus the thumb in this function.

Currently if you click anywhere on the rail and then use arrow keys to move slider, it won't work.

Reply:
Should work now.